docs/adrs/0076-partial-update-patches.md

```diff
@@ -0,0 +1,146 @@
+
+Such a design would have many advantages in addition to solving the problem of partial model updates and setting fields to null. The UpdateMTOShipment struct provides clarity by broadcasting intent: it clearly exists to update a shipment and it tells the service exactly what it wants. It also helps to further disentangle the handler layer from the database and services; since the handler now uses the MTOShipmentUpdate model, it no longer needs to know about the database structure. This design will also allow for a thorough cleanup of a lot of the handler and service logic through removal of repeated null checking and shifting business logic responsibilities cleanly to the services.
+
+### Implementation
+
+Due to the scope of work required, implementating this pattern would need to be done slowly.
+
+#### Decisions to finalize
+* Decide on a naming convention for intermediary update models
+* Determine if all models should fall under this pattern or only update models.
+
+#### Refactor stages
+* Identify and refactor any patch endpoints that require nullable fields for deleting values.
+* Fully refactor related validator methods to cleanly handle intermediary update models
+* Refactor all update models to remove dependencies on the `models` package
+* Relocate fully refactored update models into a separate package for better separation of code
+
+
+## Considered Alternatives
+* *Introduce a new struct that serves between the handler and service layers and broadcasts intent*
+* *Update model to store information about "presence"*
+* *Pass payload to service*
+* *Do nothing*
+
+## Decision Outcome
+
+### Chosen Alternative: *Introduce a new struct that serves between the handler and service layers and broadcasts intent*
+
+* `+` *The strategy for displaying details for move history rows would be much more flexible*
+* `+` *Displaying move history details would more closely resemble more conventional React-like patterns*
+* `+` *Refactoring the existing event templates could be broken down and addressed incrementally.*
+* `-` *Refactoring the existing event templates would be a sizeable amount of work.*
+* `-` *It does not solve the issue of moving events matching to multiple templates. However, it does give the flexibility to share the same template*
+* `-` *Composite struct models have dependencies on current `models` package which means they cannot be separated into a new package without introducing circular dependencies*
+
+## Pros and Cons of the Alternatives
+
+### *Update model to store information about "presence"*
+* `+` *Addresses issue of not being able to delete fields with patches*
+* `+` *Little added effort when updating model with additional fields*
+* `-` *Violates expectations of pop database models*
+* `-` *Large amount of work involved in refactoring current models and all services that use them*
+
```

Comment:
So there was no good way to make a model that would work for pop and support presence information?

Or there is but the lift is too great?

Reply:
In a way it's kind of both, but I think the "no good way to do it" argument is most compelling.
There's no good way to make a model to support presence information because presence information is beyond the scope of a database model. From what I understand, a DB model should just give information about the current state of a row in a DB at a certain point in time. What we're trying to capture here is more related to communicating information about presence from the handler to the service, which is the intention of the Update models.